pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -2290,6 +2290,9 @@ class Timedelta(_Timedelta):
         div = other // self
         return div, other - div * self
 
+    def __round__(self, freq):
```

Comment:
Doesn’t the dunder method not take a freq keyword?

Reply:
You're right it doesn't, it instead takes `ndigits` which is not relevant here, so I thought for extra customizability I'll take `freq` in place of `ndigits` (same as `NDFrame.__round__` which takes `decimals` instead of `ndigits`)

Reply:
But I have 1 question, should `freq` have a default value? as you would usually expect `round(object)` to work without any extra arguments, if so, what should the default value be?

Reply:
I would not expect that to work

Reply:
Weirdly enough it does, for example:
```python
In [1]: import pandas as pd
In [2]: td = pd.Timedelta(days=1, hours=2, minutes=34, seconds=56.78)
In [3]: round(td, "s")
Out[3]: Timedelta('1 days 02:34:57')
In [4]: round(td)
TypeError: __round__() takes exactly 2 positional arguments (1 given)

Reply:
Or were you replying to my last sentence when I said that I expect `round(object)` to work without any args

Reply:
> Or were you replying to my last sentence when I said that I expect round(object) to work without any args

That

Reply:
If this just happens to work due to an implementation detail in CPython, then it seems to me to be an abuse of the Python API and could break in the future. Is supporting other arguments in Python's `round` documented somewhere?

Reply:
[python round](https://docs.python.org/3/reference/datamodel.html#object.__round__)